File: python/src/fogbugz_mcp/tools/resolve_case.py
from __future__ import annotations

from typing import TYPE_CHECKING, Any

from mcp.server.fastmcp import FastMCP

if TYPE_CHECKING:
    from ..fogbugz_client import FogBugzClient


def register_resolve_case(mcp: FastMCP, client: FogBugzClient) -> None:
    @mcp.tool(
        name="resolve_case",
        description="Resolve an active FogBugz case. Optionally specify a resolution status and comment.",
        annotations={"destructiveHint": True},
    )
    async def resolve_case(
        case_number: int, status: str | None = None, comment: str | None = None
    ) -> str:
        """Resolve an active FogBugz case."""
        options: dict[str, Any] = {}
        if status:
            options["sStatus"] = status
        if comment:
            options["sEvent"] = comment

        result = await client.resolve(case_number, options)
        case_url = client.get_case_url(result["case"]["ixBug"])
        status_msg = f' as "{status}"' if status else ""
        return f"Case {result['case']['ixBug']} resolved{status_msg}.\nLink: {case_url}"

File: ts/src/tools/resolve-case.ts
import { z } from "zod";
import type { McpServer } from "@modelcontextprotocol/sdk/server/mcp.js";
import type { FogBugzClient } from "../fogbugz-client.js";

export function registerResolveCase(
  server: McpServer,
  client: FogBugzClient,
) {
  server.tool(
    "resolve_case",
    "Resolve an active FogBugz case. Optionally specify a resolution status and comment.",
    {
      case_number: z.number().describe("The FogBugz case number"),
      status: z
        .string()
        .optional()
        .describe("Resolution status name (e.g. 'Fixed', 'Won't Fix', 'Duplicate')"),
      comment: z
        .string()
        .optional()
        .describe("Optional comment to add when resolving"),
    },
    { destructiveHint: true },
    async ({ case_number, status, comment }) => {
      const options: { sStatus?: string; sEvent?: string } = {};
      if (status) options.sStatus = status;
      if (comment) options.sEvent = comment;

      const result = await client.resolve(case_number, options);
      const caseUrl = client.getCaseUrl(result.case.ixBug);

      return {
        content: [
          {
            type: "text",
            text: `Case ${result.case.ixBug} resolved${status ? ` as "${status}"` : ""}.\nLink: ${caseUrl}`,
          },
        ],
      };
    },
  );
}
